test/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,102 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group('assertion-in-box-wait-limbo-acked')
+--
+-- gh-9235:
+-- Assertion in box_wait_limbo_acked.
+--
+local wait_timeout = 10
+
+local function wait_pair_sync(server1, server2)
+    -- Without retrying it fails sometimes when vclocks are empty and both
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        server1:wait_for_vclock_of(server2)
+        server2:wait_for_vclock_of(server1)
+        server1:assert_follows_upstream(server2:get_instance_id())
+        server2:assert_follows_upstream(server1:get_instance_id())
+    end)
+end
+
+local function server_wait_wal_is_blocked(server)
+    server:exec(function(wait_timeout)
+        t.helpers.retrying({timeout = wait_timeout}, function()
+            t.assert(box.error.injection.get('ERRINJ_WAL_DELAY'))
+        end)
+    end, {wait_timeout})
+end
+
+local function server_wait_synchro_queue_len_is_equal(server, expected)
+    server:exec(function(expected, wait_timeout)
+        t.helpers.retrying({timeout = wait_timeout}, function(expected)
+            t.assert_equals(box.info.synchro.queue.len, expected)
+        end, expected)
+    end, {expected, wait_timeout})
+end
+
+g.before_each(function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.cluster = cluster:new({})
+    local box_cfg = {
+        replication = {
+            server.build_listen_uri('master', cg.cluster.id),
+            server.build_listen_uri('replica', cg.cluster.id),
+        },
+        election_mode = 'candidate',
+        replication_synchro_quorum = 2,
+        replication_synchro_timeout = 100000,
```

Comment:
Please, add `replication_timeout = 0.1`.

It makes the test run significantly faster without making it any more flaky.

Reply:
Although after I added `replication_timeout = 0.1` locally, one of 100 runs failed like this:
```
[015] Test failed! Output from reject file /tmp/t/rejects/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked.reject:
[015] Tarantool version is 3.0.0-alpha3-226-g08530ff54
[015] TAP version 13
[015] 1..1
[015] # Started on Fri Nov 24 15:30:02 2023
[015] # Starting group: assertion-in-box-wait-limbo-acked
[015] not ok 1	assertion-in-box-wait-limbo-acked.test_assert_last_entry_lsn_is_positive
[015] #   ...atest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua:34: expected: 1, actual: 0
[015] #   stack traceback:
[015] #   	...atest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua:32: in function 'server_wait_synchro_queue_len_is_equal'
[015] #   	...atest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua:89: in function 'assertion-in-box-wait-limbo-acked.test_assert_last_entry_lsn_is_positive'
[015] #   	...
[015] #   	[C]: in function 'xpcall'
```

But I think this can be dealt with.

Reply:
I fixed it by adding `election_fencing_mode='off'`.
```patch
diff --git a/test/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua b/test/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua
index 695d31098..d0933492d 100644
--- a/test/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua
+++ b/test/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua
@@ -46,6 +46,8 @@ g.before_each(function(cg)
             server.build_listen_uri('replica', cg.cluster.id),
         },
         election_mode = 'candidate',
+        replication_timeout = 0.1,
+        election_fencing_mode='off',
         replication_synchro_quorum = 2,
         replication_synchro_timeout = 100000,
     }
```